File: android/AllInOneJava/app/src/main/java/com/example/allinonejava/AllInOneViewActivity.java
package com.example.allinonejava;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import com.trimblemaps.navigation.core.TrimbleMapsNavigationProvider;
import com.trimblemaps.navigation.core.TrimbleMapsTripProvider;
import com.trimblemaps.navigation.ui.components.allinone.AllInOneView;
import com.trimblemaps.navigation.ui.components.allinone.AllInOneViewOptions;
import com.trimblemaps.navigation.ui.listeners.NavigationListener;
import org.jetbrains.annotations.Nullable;


public final class AllInOneViewActivity extends AppCompatActivity {
    private AllInOneView allInOneView;
    private final NavigationListener navigationCallback = new NavigationListener() {
        public void onCancelNavigation() {
            finish();
            restartActivity();
        }

        public void onNavigationFinished() {
            finish();
            restartActivity();
        }

        public void onNavigationRunning() {
        }
    };

    public final void restartActivity() {
        this.overridePendingTransition(0, 0);
        this.startActivity(getIntent());
        this.overridePendingTransition(0, 0);
    }

    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.setContentView(R.layout.activity_all_in_one_view);

        AllInOneViewOptions options =  AllInOneViewOptions.builder(this)
                .navigationListener(navigationCallback)
                .shouldSimulateRoute(true)
                .enableVanishingRouteLine(true)
                .build();

        allInOneView = findViewById(R.id.allInOneView);
        allInOneView.setAllInOneViewOptions(options);
        allInOneView.onCreate(savedInstanceState);
    }

    private long pressedTime;
    public void onBackPressed() {
        if (pressedTime + 2000 > System.currentTimeMillis()) {
            Intent intent = new Intent("android.intent.action.MAIN");
            intent.addCategory("android.intent.category.HOME");
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            startActivity(intent);
            finish();
        } else {
            Toast.makeText(this, "Press back again to exit", Toast.LENGTH_SHORT).show();
        }

        pressedTime = System.currentTimeMillis();
    }

    public void onResume() {
        super.onResume();
        allInOneView.onResume();
    }

    public void onPause() {
        super.onPause();
        allInOneView.onPause();
    }

    protected void onStart() {
        super.onStart();
        allInOneView.onStart();
    }

    protected void onStop() {
        super.onStop();
        allInOneView.onStop();
    }

    public void onLowMemory() {
        super.onLowMemory();
        allInOneView.onLowMemory();
    }

    protected void onDestroy() {
        super.onDestroy();
        allInOneView.onDestroy();
    }
}

File: AllInOneKotlin/app/src/main/java/com/example/allinonekotlin/AllInOneViewActivity.kt
package com.example.allinonekotlin

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.trimblemaps.navigation.core.TrimbleMapsNavigationProvider
import com.trimblemaps.navigation.core.TrimbleMapsTripProvider
import com.trimblemaps.navigation.ui.components.allinone.AllInOneView
import com.trimblemaps.navigation.ui.components.allinone.AllInOneViewOptions
import com.trimblemaps.navigation.ui.listeners.NavigationListener

class AllInOneViewActivity : AppCompatActivity() {

    private lateinit var allInOneView: AllInOneView

    // These are callbacks from the all-in-one navigation view
    private val navigationCallback = object : NavigationListener {
        override fun onCancelNavigation() {
            finish()
            restartActivity()
        }

        override fun onNavigationFinished() {
            finish()
            restartActivity()
        }

        override fun onNavigationRunning() {
            //
        }
    }

    fun restartActivity() {
        overridePendingTransition(0, 0);
        startActivity(getIntent())
        overridePendingTransition(0, 0);
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_all_in_one_view)

        // Create options and pass to all-in-one view
        val optionsBuilder = AllInOneViewOptions.builder(this).apply {
            navigationListener(navigationCallback)
            shouldSimulateRoute(true)
            enableVanishingRouteLine(true)
        }

        allInOneView = findViewById<AllInOneView>(R.id.allInOneView).apply {
            setAllInOneViewOptions(optionsBuilder.build())
            onCreate(savedInstanceState)
        }
    }

    private var pressedTime: Long = 0
    override fun onBackPressed() {
        if (pressedTime + 2000 > System.currentTimeMillis()) {
            val intent = Intent(Intent.ACTION_MAIN)
            intent.addCategory(Intent.CATEGORY_HOME)
            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
            startActivity(intent)
        } else {
            Toast.makeText(this, "Press back again to exit", Toast.LENGTH_SHORT).show();
        }
        pressedTime = System.currentTimeMillis();
    }

    public override fun onResume() {
        super.onResume()
        allInOneView.onResume()
    }

    public override fun onPause() {
        super.onPause()
        allInOneView.onPause()
    }

    override fun onStart() {
        super.onStart()
        allInOneView.onStart()
    }

    override fun onStop() {
        super.onStop()
        allInOneView.onStop()
    }

    override fun onLowMemory() {
        super.onLowMemory()
        allInOneView.onLowMemory()
    }

    override fun onDestroy() {
        super.onDestroy()
        allInOneView.onDestroy()
        // Delete any existing instances of TrimbleMapsNavigation and TrimbleMapsTrip
        TrimbleMapsNavigationProvider.destroy()
        TrimbleMapsTripProvider.destroy()
    }
}